pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -352,29 +367,36 @@ func (m *mergeQuerier) Select(ctx context.Context, sortSeries bool, hints *stora
 	matchedIDs, filteredMatchers := FilterValuesByMatchers(m.idLabelName, ids, matchers...)
 
 	jobs := make([]string, 0, len(matchedIDs))
-	seriesSets := make([]storage.SeriesSet, len(matchedIDs))
 	for id := range matchedIDs {
 		jobs = append(jobs, id)
 	}
 
+	return m.innerSelectFunc(ctx, jobs, start, m.upstreamQueryWaitDuration, m.upstream, m.idLabelName, m.maxConcurrency, sortSeries, hints, filteredMatchers...)
+}
+
+type InnerSelectFunc func(ctx context.Context, jobs []string, start time.Time, upstreamQueryWaitDuration prometheus.Histogram, upstream MergeQuerierUpstream, idLabelName string, maxConcurrency int, sortSeries bool, hints *storage.SelectHints, filteredMatchers ...*labels.Matcher) storage.SeriesSet
```

Comment:
let's come up with a better name. Currently inner refers to the fact that it happens within another select func, but in reality there are a dozen nested select funcs. How about `MultiTenantSelectFunc` (the package is called tenantfederation, so this func is supposed to run the select for multiple tenants)?


In a godoc can you also explain what this function is supposed to be used for and what some of the parameters mean? like `jobs` is very unclear, how is `idLabelName` intended to be used and that `sortSeries, hints, filteredMatchers` are there only to be passed to the `upstream`

Also can we call `filteredMatchers` `matchers`?